browser/extensions/api/brave_shields_api.cc

```diff
@@ -4,13 +4,42 @@
 
 #include "brave/browser/extensions/api/brave_shields_api.h"
 
+#include "brave/common/extensions/api/brave_shields.h"
+#include "brave/content/common/frame_messages.h"
+#include "chrome/browser/extensions/api/tabs/tabs_constants.h"
+#include "chrome/browser/extensions/extension_tab_util.h"
+#include "chrome/browser/profiles/profile.h"
+#include "content/public/browser/web_contents.h"
+
 namespace extensions {
 namespace api {
 
-BraveShieldsDummyFunction::~BraveShieldsDummyFunction() {
+BraveShieldsAllowScriptsOnceFunction::~BraveShieldsAllowScriptsOnceFunction() {
 }
 
-ExtensionFunction::ResponseAction BraveShieldsDummyFunction::Run() {
+ExtensionFunction::ResponseAction BraveShieldsAllowScriptsOnceFunction::Run() {
+  std::unique_ptr<brave_shields::AllowScriptsOnce::Params> params(
+      brave_shields::AllowScriptsOnce::Params::Create(*args_));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  // Get web contents for this tab
+  content::WebContents* contents = nullptr;
+  if (!ExtensionTabUtil::GetTabById(
+        params->tab_id,
+        Profile::FromBrowserContext(browser_context()),
+        include_incognito(),
+        nullptr,
+        nullptr,
+        &contents,
+        nullptr)) {
+    return RespondNow(Error(tabs_constants::kTabNotFoundError,
+                            base::IntToString(params->tab_id)));
+  }
+
+  // Send IPC msg to update the list in this tab's content settings observers
+  contents->SendToAllFrames(
+      new BraveFrameMsg_AllowScriptsOnce(MSG_ROUTING_NONE, params->origins));
```

Comment:
Noted there is an issue here that frames which might be created after this will not get this list.
Probably could be solved by saving the list (which is specific to one webcontent) in the browser process (maybe in BraveWebContentsObserver or BraveTabSpecificContentSetting) and send the list in RenderFrameCreated to those subsequent frames that are created later.

Reply:
addressed by https://github.com/brave/brave-core/pull/97/commits/00cfdb3553fd83b6fb2ef436ded8ed880b32d9fc